proxy-robots-module/src/main/java/orwell/proxy/Tank.java

```diff
@@ -0,0 +1,331 @@
+    private String teamName;
+
+    private final TankDeltaState tankDeltaState = new TankDeltaState();
+
+    public Tank(String bluetoothName, String bluetoothID, ICamera camera,
+                MessageFramework mf, String image) {
+        setBluetoothName(bluetoothName);
+        setBluetoothID(bluetoothID);
+        this.camera = camera;
+        nxtInfo = new NXTInfo(NXTCommFactory.BLUETOOTH, bluetoothName,
+                bluetoothID);
+        mfTank = mf;
+        mf.addMessageListener(this);
+        this.image = image;
+    }
+
+    public Tank(String bluetoothName, String bluetoothID, ICamera camera, String image) {
+        this(bluetoothName, bluetoothID, camera, new MessageFramework(), image);
+    }
+
+    @Override
+    public void buildRegister() {
+        registerBuilder.setTemporaryRobotId(routingID);
+        registerBuilder.setVideoUrl(camera.getUrl());
+        registerBuilder.setImage(image);
+        if ("" == image) {
+            logback.info("Image of tank " + routingID + " is empty. "
+                    + "This will probably be an issue for the serverGame");
+        }
+        register = registerBuilder.build();
+    }
+
+    private Register getRegister() {
+        if (null == register) {
+            buildRegister();
+        }
+        return register;
+    }
+
+    @Override
+    public byte[] getRegisterBytes() {
+        Register register = getRegister();
+        if (null != register)
+            return register.toByteArray();
+        else
+            return null;
+    }
+
+    @Override
+    public Input getControllerInput() {
+        return currentControllerInput;
+    }
+
+    @Override
+    public void setControllerInput(byte[] inputMessage) {
+        try {
+            this.currentControllerInput = Input.parseFrom(inputMessage);
+            if (currentControllerInput.hasMove()) {
+                String payloadMove = "input move ";
+                payloadMove += currentControllerInput.getMove().getLeft() + " "
+                        + currentControllerInput.getMove().getRight();
+                UnitMessage msg = new UnitMessage(UnitMessageType.Command,
+                        payloadMove);
+                mfTank.SendMessage(msg);
+            }
+            if (currentControllerInput.hasFire()
+                    && (currentControllerInput.getFire().getWeapon1() || currentControllerInput
+                    .getFire().getWeapon2())) {
+                String payloadFire = "input fire ";
+                payloadFire += currentControllerInput.getFire().getWeapon1()
+                        + " " + currentControllerInput.getFire().getWeapon2();
+                UnitMessage msg = new UnitMessage(UnitMessageType.Command,
+                        payloadFire);
+                mfTank.SendMessage(msg);
+            }
+        } catch (InvalidProtocolBufferException e) {
+            // TODO Auto-generated catch block
```

Comment:
This is a //TODO line automatically added when you create a try/catch block.
I'll remove the TODO and add a logback.error()